src/index.ts

```diff
@@ -293,17 +297,34 @@ export const createClaimWithChildInstance = (
       pauseStatus: boolean,
       signer: ethers.Signer
     ): Promise<ethers.ContractTransaction> => {
-      const contract = await getAirWild2SaleContract(
-        signer,
-        config.contractAddress
-      );
-      const tx = await airWildS2Actions.setPauseStatus(
+      const contract = await getClaimContract(signer, config.contractAddress);
+      const tx = await claimWithChildSaleActions.setPauseStatus(
         pauseStatus,
         contract,
         signer
       );
       return tx;
     },
+    getClaimingIDsForUser: async (walletID: string): Promise<string[]> => {
+      const claimingToken = await getClaimingToken(
+        config.web3Provider,
+        config.claimingRegistrarAddress
+      );
+      let childDomain: ethers.BigNumber;
+      const claimingIDs: string[] = [];
+      const bigNumDomainsOwned = await claimingToken.balanceOf(walletID);
+      const numDomainsOwned = bigNumDomainsOwned.toNumber();
+      if (numDomainsOwned != 0) {
+        let i = 0;
+        while (i < numDomainsOwned) {
```

Comment:
You don't increment the value of `i` here. Maybe you meant `while (claimingIDs.length < numDomainsOwned)` ? If so delete `let i = 0` and update this condition

Reply:
great catch